src/Domain/DomainCreators/Sphere.cpp

```diff
@@ -0,0 +1,112 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/DomainCreators/Sphere.hpp"
+
+#include <array>
+#include <unordered_set>
+#include <vector>
+
+#include "Domain/Block.hpp"
+#include "Domain/BlockNeighbor.hpp"
+#include "Domain/CoordinateMaps/AffineMap.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/Wedge3D.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/MakeVector.hpp"
+
+namespace DomainCreators {
+
+Sphere::Sphere(typename InnerRadius::type inner_radius,
+               typename OuterRadius::type outer_radius,
+               typename InitialRefinement::type initial_refinement,
+               typename InitialGridPoints::type initial_number_of_grid_points,
+               typename WithEquiangularMap::type with_equiangular_map) noexcept
+    // clang-tidy: trivially copyable
+    : inner_radius_(std::move(inner_radius)),                    // NOLINT
+      outer_radius_(std::move(outer_radius)),                    // NOLINT
+      initial_refinement_(                                       // NOLINT
+          std::move(initial_refinement)),                        // NOLINT
+      initial_number_of_grid_points_(                            // NOLINT
+          std::move(initial_number_of_grid_points)),             // NOLINT
+      with_equiangular_map_(std::move(with_equiangular_map)) {}  // NOLINT
+
+Domain<3, Frame::Inertial> Sphere::create_domain() const noexcept {
+  using Wedge3DMap = CoordinateMaps::Wedge3D;
+  using Wedge3DType =
+      CoordinateMap<Frame::Logical, Frame::Inertial, Wedge3DMap>;
+  using AffineMap = CoordinateMaps::AffineMap;
+  using AffineMap3D =
+      CoordinateMaps::ProductOf3Maps<AffineMap, AffineMap, AffineMap>;
+
+  std::vector<std::array<size_t, 8>> corners{
+      {{7, 5, 8, 6, 15, 13, 16, 14}},  // Upper z
```

Comment:
I think the blocks should have some ordering, either with respect to direction or axis.